RFS/src/main/java/org/opensearch/migrations/bulkload/common/OpenSearchClientFactory.java

```diff
@@ -20,21 +19,20 @@
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 import reactor.core.publisher.Mono;
-import reactor.util.retry.Retry;
 
 @Getter
 @Slf4j
 public class OpenSearchClientFactory {
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
     private ConnectionContext connectionContext;
     private Version version;
-    protected final RestClient client;
-    protected static final ObjectMapper objectMapper = new ObjectMapper();
-
-    public OpenSearchClientFactory(RestClient client) {
-        this.client = client;
-    }
-
+    RestClient client;
```

Comment:
Made this 'package internal' so the tests can change the client that was created for a mocked version.